theano/gpuarray/extra_ops.py

```diff
@@ -430,8 +430,8 @@ def c_support_code_struct(self, node, nodename):
                         size_t dimBlock[3] = {1, 1, 1};
                         size_t tmp0 = shape[axis]-2;
                         size_t tmp1 = shape[axis]-1;
-                        void* kernel_params[] = {(void*) input->ga.data,
-                                                 (void*) output->ga.data,
+                        void* kernel_params[] = {(void*) ((char *)(input->ga.data) + input->ga.offset),
```

Comment:
@lamblin Related to this: https://github.com/mila-udem/support/issues/65, where the problem is a `gpudata` with an invalid `ctx`.

Why is `(void*) ((char *)(input->ga.data) + input->ga.offset` at all valid?

From my understanding of the code, `input` is `PyGpuArrayObject*`, `input->ga` is `GpuArray*` and `input->ga.data` is therefore `gpudata*`. So why on earth would one add a byte offset to it?